core/src/host_imports/imports.rs

```diff
@@ -62,16 +61,22 @@ pub(crate) fn create_host_import_func(
     let raw_args = itypes_args_to_wtypes(&argument_types);
     let raw_output = itypes_output_to_wtypes(&output_type_to_types(output_type));
 
-    let func = move |ctx: &mut Ctx, inputs: &[WValue]| -> Vec<WValue> {
+    let func = move |mut caller: <WB as WasmBackend>::Caller<'_>,
```

Comment:
I'd suggest to move this lambda body to separate function, and the lambda to have something like
```
|caller: ..., inputs: ...| the_func(caller, inputs, captured1, captured2, ...)
```